### PR TITLE
feat(extension): Integration of claim compliance provider

### PR DESCRIPTION
**Pull request by robertschubert**
## What this PR changes/adds

CCP was integrated into this component.

## Why it does that

If there is no claimComplianceProviderResponse in the payload this component calls the claim compliance provider and stores the response in the asset. 
On ContractDefinitionCreation the responses are collected from all assets and the result is being written on the blockchain / ipfs, so it can be picked up by the blockchain listener (or any other component).

## Further notes

Also contains some changes in the build @drandarov-io made to make the build running in a azure k8s cluster

## Linked Issue(s)

-

## Checklist

- [x] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [x] documented code?
- [ ] assigned appropriate label?
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-edc/Samples/blob/main/CONTRIBUTING.md#submit-a-pull-request) for details_)

### CODE REVIEW

**Comment by JulianLegler** [comment]
You can create Assets without the new properties your pull request introduced but then later on during ContractDefinition creation the ContractDefinition will not be created and the process is interrupted because of :
```
2024-07-24T14:01:43.735Z INFO '[BlockchainContractCreator] ContractDefinition: for '5c5707e9-b631-41e2-a0c1-e20dffc92d02' and '5c5707e9-b631-41e2-a0c1-e20dffc92d02' targeting 'not implemented' created in EDC, start now with Blockchain related steps ...' : 898d06f2d3b912d48e4459591c93cc53 efc1305e804cde23 [scopeInfo: org.eclipse.edc.monitor.logger.LoggerMonitor:] {}
2024-07-24T14:01:43.735Z INFO '[BlockchainContractCreator] formating POJO to JSON ...' : 898d06f2d3b912d48e4459591c93cc53 efc1305e804cde23 [scopeInfo: org.eclipse.edc.monitor.logger.LoggerMonitor:] {}
2024-07-24T14:01:43.737Z WARN 'No CCP-Response found for Asset with id test-document-53' : 898d06f2d3b912d48e4459591c93cc53 efc1305e804cde23 [scopeInfo: org.eclipse.edc.monitor.logger.LoggerMonitor:] {}
2024-07-24T14:01:43.737Z WARN 'Something went wrong during the Verifiable Presentation creation / CCP response extraction for the Contract with id 26' : 898d06f2d3b912d48e4459591c93cc53 efc1305e804cde23 [scopeInfo: org.eclipse.edc.monitor.logger.LoggerMonitor:] {}
```

Can you check what's happening there? It seams as there is logic to handle this case there seems to be still an error.
We don't have to keep backwards capabilities. So we could, in the first place, interrupt the Asset creation if these properties are missing, or we fix this issues if we are fine that there are contracts that dont have the CCP correctly set.

I created the Assets with this JSON:
```json
{
    "@context": {
        "edc": "https://w3id.org/edc/v0.0.1/ns/"
    },
    "@id": "{{assetNameFull}}",
    "@type": "https://w3id.org/edc/v0.0.1/ns/Asset",
    "https://w3id.org/edc/v0.0.1/ns/properties": {
        "https://w3id.org/edc/v0.0.1/ns/name": "{{assetNameFull}}",
        "https://w3id.org/edc/v0.0.1/ns/description": "TestAssetDescription",
        "https://w3id.org/edc/v0.0.1/ns/id": "{{assetNameFull}}",
        "https://w3id.org/edc/v0.0.1/ns/version": "0.2.0",
        "https://w3id.org/edc/v0.0.1/ns/contenttype": "text/plain"
    },
    "https://w3id.org/edc/v0.0.1/ns/dataAddress": {
        "@type": "https://w3id.org/edc/v0.0.1/ns/DataAddress",
        "name": "Test asset",
        "baseUrl": "https://jsonplaceholder.typicode.com/users",
        "https://w3id.org/edc/v0.0.1/ns/type": "HttpData"
    }
}
```
And the ContractDefinition:
```json
{
    "@context": { "edc": "https://w3id.org/edc/v0.0.1/ns/" },
    "@id": "{{contractDefinitionId}}",
    "@type": "https://w3id.org/edc/v0.0.1/ns/ContractDefinition",
    "https://w3id.org/edc/v0.0.1/ns/accessPolicyId": "{{policyUID}}",
    "https://w3id.org/edc/v0.0.1/ns/contractPolicyId": "{{policyUID}}",
    "https://w3id.org/edc/v0.0.1/ns/assetsSelector": [
        {
            "@type": "https://w3id.org/edc/v0.0.1/ns/Criterion",
            "https://w3id.org/edc/v0.0.1/ns/operandLeft": "https://w3id.org/edc/v0.0.1/ns/id",
            "https://w3id.org/edc/v0.0.1/ns/operator": "=",
            "https://w3id.org/edc/v0.0.1/ns/operandRight": "{{assetNameFull}}"
        }
    ]
}
```



The intended way, with using the new asset properties, seems to work correctly. I tested it from new asset creation over contract negotiation and data transmission without an error.

Also, the refactoring to an own Extension is, as discussed earlier, postponed until the new project structure is merged, which would be my next task after this PR is accepted, so the overall structure of this PR is fine to me.

**Comment by robertschubert** [comment]
Hi @JulianLegler 
the logic is (in BlockchainContractCreator):

If URL for ccp is configured, the CCP-Response is expected in the asset's properties.
But I agree: we could write into the blockchain even if there is no asset with a CCP-Response. Do you think we should call the legacy way in this case (which is calling the SD-Creator)? 

We should discuss in general how long we should keep the legacy way (calling SD-Creator). Technically it's fine and we can show hat a verifiablePresentation is written to the blockchain / ipfs. But it makes no sence in functional matter because the verifiablePresentation is not describing the asset (it is no metadata).

So I would just write something like
```
{
                    "edcContractdefinition": $jsonRepresentationOfContractDefinition
}
```
and not
```
{
                    "edcContractdefinition": $jsonRepresentationOfContractDefinition ,
                    "claimComplianceProviderResponses": $verifiablePresentationOfContract 
}
```
The blockchain-listener of course needs to handle this case also, but i will take care of.

But i would not recommend to remove the SD-Creator stuff in this PR because of the increased risk of merge conflicts.

Long story short: 
I will change the code so the data is written to the blockchain in the legacy way. after that will remove this stuff seperately.

**Comment by robertschubert** [comment]
@JulianLegler 
I fixed the code. Writing to blockchain should happen now if you have an asset without CCP response.

**Comment by robertschubert** [comment]
> @JulianLegler I fixed the code. Writing to blockchain should happen now if you have an asset without CCP response.

obviously forgot to push this commit. Will do it until Monday.

**Comment by robertschubert** [comment]
I have pushed my changes so writing into the blockchain will also work when there is no CCP response.

**Review by JulianLegler — APPROVED** [review]
